matchms/similarity/BaseSimilarity.py

```diff
@@ -231,13 +296,19 @@ def _sparse_array_with_mask_without_filter(self, references: Sequence[SpectrumTy
             scores[i] = self.pair(references[i_row], queries[i_col])
         return COOMatrix(row_idx=mask_indices.idx_row, column_idx=mask_indices.idx_col, scores=scores, scores_dtype=self.score_datatype)
 
-    def _sparse_array_with_filter(self, references: Sequence[SpectrumType], queries: Sequence[SpectrumType],
-                                 mask_indices: COOIndex) -> COOMatrix:
-        """Uses a mask to compute only the required scores and filters scores that do not pass the filter.
+    def _sparse_array_with_filter(
```

Comment:
Maybe we should name this to _sparse_array_with_filter_with_mask, to follow consistent naming.

Reply:
Should we then also rename `_matrix_with_mask` to `_matrix_with_filter_with_mask` to make it more consistent?

Reply:
Yes indeed! 

And maybe also fix the ordering, while we are at it. Now we start with mask most of the time, but a few cases start with filter instead. 

So to change is:
_sparse_array_with_filter_without_mask -> _sparse_array_without_mask_with_filter
_sparse_array_with_filter -> _sparse_array_with_mask_with_filter
_matrix_with_mask -> _matrix_with_mask_with_filter

Reply:
Done